bk://kernel.bkbits.net/gregkh/linux/driver-2.6

# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/06/22 16:24:54-07:00 akpm@osdl.org 
#   [PATCH] raw.c cleanups
#   
#   - pass the raw_config_request by reference, not by value.
#   
#   - fix whitespace drainbamage
#   
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Greg Kroah-Hartman <greg@kroah.com>
# 
# drivers/char/raw.c
#   2004/06/22 08:01:30-07:00 akpm@osdl.org +8 -7
#   raw.c cleanups
# 
diff -Nru a/drivers/char/raw.c b/drivers/char/raw.c
--- a/drivers/char/raw.c	2004-06-28 19:14:59 -07:00
+++ b/drivers/char/raw.c	2004-06-28 19:14:59 -07:00
@@ -125,11 +125,11 @@
 	return ioctl_by_bdev(bdev, command, arg);
 }
 
-static void bind_device(struct raw_config_request rq)
+static void bind_device(struct raw_config_request *rq)
 {
-	class_simple_device_remove(MKDEV(RAW_MAJOR, rq.raw_minor));
-	class_simple_device_add(raw_class, MKDEV(RAW_MAJOR, rq.raw_minor),
-				      NULL, "raw%d", rq.raw_minor);
+	class_simple_device_remove(MKDEV(RAW_MAJOR, rq->raw_minor));
+	class_simple_device_add(raw_class, MKDEV(RAW_MAJOR, rq->raw_minor),
+				      NULL, "raw%d", rq->raw_minor);
 }
 
 /*
@@ -200,15 +200,16 @@
 			if (rq.block_major == 0 && rq.block_minor == 0) {
 				/* unbind */
 				rawdev->binding = NULL;
-				class_simple_device_remove(MKDEV(RAW_MAJOR, rq.raw_minor));
+				class_simple_device_remove(MKDEV(RAW_MAJOR,
+								rq.raw_minor));
 			} else {
 				rawdev->binding = bdget(dev);
 				if (rawdev->binding == NULL)
 					err = -ENOMEM;
 				else {
 					__module_get(THIS_MODULE);
-					bind_device(rq);
-					}
+					bind_device(&rq);
+				}
 			}
 			up(&raw_mutex);
 		} else {
